controllers/cstates_controller_test.go

```diff
@@ -149,6 +172,426 @@ func TestCStates_Reconcile(t *testing.T) {
+			powerProfileSetup:      twoProfileSetup,
+			withSharedPoolError:    true,
+			reconcileMethodAtError: controllerRestoreMethodName,
+			expectError:            true,
+			reconcileErrorMessages: []string{
+				fmt.Sprintf(sharedPoolErrorMessage, controllerRestoreMethodName),
+			},
+		},
+		{
+			name:                "Test 5 - restore fails with shared &  pool errors",
+			cStatesSetup:        cStatesSetup,
+			powerNodesSetup:     twoPowerNodeSetup,
+			powerProfileSetup:   twoProfileSetup,
+			withSharedPoolError: true,
+			exclusivePoolsWithError: map[string]bool{
+				"performance":         true,
+				"balance-performance": true},
+			reconcileMethodAtError: controllerRestoreMethodName,
+			expectError:            true,
+			reconcileErrorMessages: []string{
+				fmt.Sprintf(exclusivePoolErrorMessage, "performance", controllerRestoreMethodName),
+				fmt.Sprintf(exclusivePoolErrorMessage, "balance-performance", controllerRestoreMethodName),
+			},
+		},
+		{
+			name:                "Test 6 - restore fails with shared & individual pool errors",
+			cStatesSetup:        cStatesSetup,
+			powerNodesSetup:     twoPowerNodeSetup,
+			powerProfileSetup:   twoProfileSetup,
+			withSharedPoolError: true,
+			individualPoolsWithError: map[string]bool{
+				"0": true, "3": true, "4": true},
+			reconcileMethodAtError: controllerRestoreMethodName,
+			expectError:            true,
+			reconcileErrorMessages: []string{
+				fmt.Sprintf(sharedPoolErrorMessage, controllerRestoreMethodName),
+				fmt.Sprintf(individualPoolErrorMessage, "0", controllerRestoreMethodName),
+				fmt.Sprintf(individualPoolErrorMessage, "3", controllerRestoreMethodName),
+				fmt.Sprintf(individualPoolErrorMessage, "4", controllerRestoreMethodName),
+			},
+		},
+		{
+			name:                   "Test 7 - apply fails with shared pool error",
+			cStatesSetup:           cStatesSetup,
+			powerNodesSetup:        twoPowerNodeSetup,
+			powerProfileSetup:      twoProfileSetup,
+			withSharedPoolError:    true,
+			reconcileMethodAtError: controllerApplyMethodName,
+			expectError:            true,
+			reconcileErrorMessages: []string{fmt.Sprintf(sharedPoolErrorMessage, controllerApplyMethodName)},
+		},
+		{
+			name:                    "Test 8 - apply fails with shared & exclusive pool errors",
+			cStatesSetup:            cStatesSetup,
+			powerNodesSetup:         twoPowerNodeSetup,
+			powerProfileSetup:       twoProfileSetup,
+			withSharedPoolError:     true,
+			exclusivePoolsWithError: map[string]bool{"balance-performance": true},
+			reconcileMethodAtError:  controllerApplyMethodName,
+			expectError:             true,
+			reconcileErrorMessages: []string{
+				fmt.Sprintf(sharedPoolErrorMessage, controllerApplyMethodName),
+				fmt.Sprintf(exclusivePoolErrorMessage, "balance-performance", controllerApplyMethodName),
+			},
+		},
+		{
+			name:                     "Test 9 - verify fails with shared, exclusive and individual pool error2",
```

Comment:
```suggestion
			name:                     "Test 9 - verify fails with shared, exclusive and individual pool errors",
```

Reply:
Will address all changes to controllers/controller tests in the upcoming release